src/components/program-settings.js

```diff
@@ -139,94 +139,64 @@ class ProgramSettings extends React.Component {
-        const valueInput = e.target.value
-        switch (e.target.name) {
+    chooseSetting = (name, value) => {
+        switch (name) {
             case 'teiDownload':
-                e.target.value > maxValues.teiDownload
-                    ? (e.target.value = maxValues.teiDownload)
-                    : (e.target.value = valueInput)
+                value > maxValues.teiDownload
+                    ? (value = parseInt(maxValues.teiDownload))
+                    : (value = parseInt(value))
                 break
             case 'teiDBTrimming':
-                e.target.value > maxValues.teiDBTrimming
-                    ? (e.target.value = maxValues.teiDBTrimming)
-                    : (e.target.value = valueInput)
+                value > maxValues.teiDBTrimming
+                    ? (value = parseInt(maxValues.teiDBTrimming))
+                    : (value = parseInt(value))
                 break
             case 'teReservedDownload':
-                e.target.value > maxValues.teReservedDownload
-                    ? (e.target.value = maxValues.teReservedDownload)
-                    : (e.target.value = valueInput)
+                value > maxValues.teReservedDownload
+                    ? (value = parseInt(maxValues.teReservedDownload))
+                    : (value = parseInt(value))
                 break
             case 'teReservedDBTrimming':
-                e.target.value > maxValues.teReservedDBTrimming
-                    ? (e.target.value = maxValues.teReservedDBTrimming)
-                    : (e.target.value = valueInput)
+                value > maxValues.teReservedDBTrimming
+                    ? (value = parseInt(maxValues.teReservedDBTrimming))
+                    : (value = parseInt(value))
                 break
             case 'eventsDownload':
-                e.target.value > maxValues.eventsDownload
-                    ? (e.target.value = maxValues.eventsDownload)
-                    : (e.target.value = valueInput)
+                value > maxValues.eventsDownload
+                    ? (value = parseInt(maxValues.eventsDownload))
+                    : (value = parseInt(value))
                 break
             case 'eventsDBTrimming':
-                e.target.value > maxValues.eventsDBTrimming
-                    ? (e.target.value = maxValues.eventsDBTrimming)
-                    : (e.target.value = valueInput)
+                value > maxValues.eventsDBTrimming
+                    ? (value = parseInt(maxValues.eventsDBTrimming))
+                    : (value = parseInt(value))
                 break
             default:
                 break
         }
+        return value
+    }
 
+    handleChange = e => {
+        e.preventDefault()
+        let { value } = e.target
+        value = this.chooseSetting(e.target.name, value)
         this.setState({
             ...this.state,
-            [e.target.name]: e.target.value,
+            [e.target.name]: value,
```

Comment:
I don't think it's necessary to use a mutable variable here, you could do this instead:

```js
        const value = this.chooseSetting(e.target.name, e.target.value)
        this.setState({
            ...this.state,
            [e.target.name]: value,
         })
```

(you could also just call `this.chooseSetting` directly on line 186, but the temporary `const` variable is fine too)

Reply:
I'm just calling `this.chooseSetting` directly on line 184